test/performance/shmem_perf_suite/common.h

```diff
@@ -14,6 +14,21 @@
 #include <string.h>
 #include <assert.h>
 #include <stdbool.h>
+#include <sys/time.h>
+
+#ifndef HAVE_SHMEMX_WTIME
+double
+shmemx_wtime(void)
+{
+    double wtime;
+    struct timeval tv;
+
+    gettimeofday(&tv, NULL);
+    wtime = tv.tv_sec;
+    wtime += (double)tv.tv_usec / 1000000.0;
+    return wtime;
+}
+#endif
```

Comment:
This returns the current time, in seconds, just like ```MPI_Wtime``.  There is a corresponding implementation of ```shmemx_wtime()``` in the SOS implementation that should be updated if we make any changes here (the code you commented on here is not actually exercised when you build this benchmark using SOS).

There is a secondary issue to the loss of precision you noted, which is that ```gettimeofday``` is a poor way to measure the passage of time (see, https://blog.habets.se/2010/09/gettimeofday-should-never-be-used-to-measure-time).  We can use the POSIX timer, ```clock_gettime``` with ```CLOCK_REALTIME``` or ```CLOCK_MONOTONIC``` when it's available and fall back to ```gettimeofday``` otherwise.

```shmemx_time``` was a quick hack from a while back.  Good timers will take some configury and system-specific hacking.  We should be conscious of scope creep..